central/cluster/util/util.go

```diff
@@ -3,15 +3,16 @@ package util
 import (
 	"context"
 
-	cluster "github.com/stackrox/rox/central/cluster/datastore"
+	"github.com/stackrox/rox/central/role/sachelper"
 	"github.com/stackrox/rox/pkg/errox"
 )
 
 // GetClusterIDFromNameOrID returns the ID of cluster. If cluster is already an ID than it is returned
-// as is, if cluster is a name will return the associated cluster's ID.
-func GetClusterIDFromNameOrID(ctx context.Context, ds cluster.DataStore, cluster string) (string, error) {
+// as is, if cluster is a name will return the associated cluster's ID. Permissions are used by the
+// sachelper to determine which clusters are accessible, see sachelper for details.
+func GetClusterIDFromNameOrID(ctx context.Context, clusterSACHelper sachelper.ClusterSacHelper, cluster string, permissions []string) (string, error) {
```

Comment:
is permissions ever not `nil`? Guess it doesn't hurt to keep the option for non-nil though

Reply:
For the current usages it is always `nil`. Initially I was passing the specific resource (ie: `Image` or `Detection`) - when I learned that `Detection` is a global permission that the `sachelper` ignores, combined with permissions already being validated by the API auth function at the point the method is called - it seemed redundant. It being a utility function I kept the option there for future needs.